napari/_qt/qt_viewer.py

```diff
@@ -217,10 +190,19 @@ def __init__(
         self._dockPerformance = None
 
         # This dictionary holds the corresponding vispy visual for each layer
-        self.layer_to_visual = {}
-        self.overlay_to_visual = {}
+        self.canvas = VispyCanvas(
+            viewer=viewer,
+            parent=self,
+            size=self.viewer._canvas_size,
+        )
 
```

Comment:
Instead of hardcoding `ViewerCanvas` as the only option add change 

```
def __init__(
        self, viewer: ViewerModel, show_welcome_screen: bool = False
    ) -> None:
````

to
```
def __init__(
        self, viewer: ViewerModel,
        show_welcome_screen: bool = False
        canvas_class: Type[VispyCanvas] = VispyCanvas
    ) -> None:
``` 
(this require import type from typing)

Then here replace `self.canvas = VispyCanvas(` with `self.canvas = canvas_class(`

Reply:
Changes implemented with [401ce8c](https://github.com/napari/napari/pull/5432/commits/401ce8ca348141f14a92ae08374ba4cd871c6aea)